### DISCUSSION TITLE
How to create units in qalc?

### DISCUSSION THREAD

**Discussion by vlad2305m** [Qalculate/libqalculate#505] [Q&A] [answered]
I am integrating qalc into a Minecraft mod, so I want to add custom units (preferably not interfering with global units configuration). I can only use command flags and run anything in qalc itself. I do not want to open the GUI.

Is there any command in qalc to create and configure units? (like `function`)
I do not want to use JNI with C calls, as it will complicate the development.

Target systems: Linux (.tar.xz), Windows (portable), maybe Mac

**Comment by hanna-kn** [comment, marked as answer]
It is currently not possible to add or modify units using qalc.

Your best option is probably to add the required units to a local definitions xml file (using the GUI, or by hand), and distribute it with the mod. Set the `QALCULATE_USER_DIR` environment variable to specify the location of the file.
